tests/Microsoft.Identity.Test.Unit/AppConfigTests/CiamAuthorityHelperTests.cs

```diff
@@ -0,0 +1,105 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using Microsoft.Identity.Client;
+using Microsoft.Identity.Client.AppConfig;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Microsoft.Identity.Test.Unit.AppConfigTests
+{
+    [TestClass]
+    public class CiamAuthorityHelperTests
+    {
+        private readonly string _ciamInstance = "https://login.ciamlogin.com";
+        private readonly string _ciamTenantGuid = "5e156ef5-9bd2-480c-9de0-d8658f21d3f7";
+        private readonly string _ciamTenant = "idgciamdemo.onmicrosoft.com";
+
+        // Possible CIAM authorities:
+        // https://login.ciamlogin.com/idgciamdemo.onmicrosoft.com
+        // https://idgciamdemo.ciamlogin.com
+        // https://login.ciamlogin.com/5e156ef5-9bd2-480c-9de0-d8658f21d3f7
+        [TestMethod]
+        public void CiamAuthorityAdapater_WithAuthorityAndNamedTenantTest()
+        {
+            // Arrange
+            string ciamAuthority = _ciamInstance + '/' + _ciamTenant;
+
+            // Act
+            CiamAuthorityHelper ciamAuthorityHelper = new CiamAuthorityHelper(new Uri(ciamAuthority));
+
+            // Assert
+            Assert.AreEqual(ciamAuthority, ciamAuthorityHelper.TransformedAuthority.AbsoluteUri);
+            Assert.AreEqual(_ciamInstance, ciamAuthorityHelper.TransformedInstance);
+            Assert.AreEqual(_ciamTenant, ciamAuthorityHelper.TransformedTenant);
+        }
+
+        [TestMethod]
+        public void CiamAuthorityAdapater_WithAuthorityAndTenantGUIDTest()
+        {
+            // Arrange
+            string ciamAuthority = _ciamInstance + '/' + _ciamTenantGuid;
+
+            // Act
+            CiamAuthorityHelper ciamAuthorityHelper = new CiamAuthorityHelper(new Uri(ciamAuthority));
+
+            // Assert
+            Assert.AreEqual(ciamAuthority, ciamAuthorityHelper.TransformedAuthority.AbsoluteUri);
+            Assert.AreEqual(_ciamInstance, ciamAuthorityHelper.TransformedInstance);
+            Assert.AreEqual(_ciamTenantGuid, ciamAuthorityHelper.TransformedTenant);
+        }
+
+        [TestMethod]
+        public void CiamAuthorityAdapater_WithCiamLoginTest()
+        {
+            // Arrange
+            string ciamAuthority = "https://idgciamdemo.ciamlogin.com/";
+            string ciamTransformedInstance = "https://login.ciamlogin.com/";
```

Comment:
Same thing, the transformed authority should not be "login."

I took the time to express what the behavior should be in this PR comment:
https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/4054#issuecomment-1492754089

I don't think that my comment had "login" anywhere?

More below too VV

Reply:
ok, I missed the login part. i was trying too hard to maintain the functionality of the POC.